### PR TITLE
Fix/trip invite format profile

### LINKED ISSUE #332: Fix the display of trip invites in profile screen

### PR DESCRIPTION
**Pull request by BogdanCDev**
## Summary

This PR introduces a **bug fix** to the profile screen. Previously, the trip invite items displayed the sender's ID and trip ID instead of their respective names. Additionally, two new methods, `getTripByID` and `getUserByID`, have been implemented to retrieve trip and user details asynchronously.

## Changes

- **Models:**
  - No changes to the existing data models.

- **Screens/UI:**
  - Updated `TripInviteItem` in the profile screen to display the trip name and sender's name instead of their IDs.

- **Logic/Controllers:**
  - Introduced methods in `TripsViewModel` and `UserViewModel` for fetching trip and user details:
    - `getTripByID`: Fetches trip details using the trip ID.
    - `getUserByID`: Retrieves user details using the user ID.

- **Bug Fixes/Improvements:**
  - Fixed the incorrect display of sender's ID and trip ID in trip invites.

## Checklist

- [x] This PR resolves #332
- [x] Tests have been added for new functionality
- [x] Screenshots or UI changes have been attached
- [x] Documentation has been updated

## Testing

**Manual Testing:**
- [x] Tested on emulator and physical device to ensure the sender’s name and trip name are correctly displayed.
- [x] Verified the functionality of `getTripByID` and `getUserByID` in both success and failure scenarios.

**Unit Tests:**
- [x] Added unit tests for `getTripByID` and `getUserByID`.

## Related Issues/Tickets
Closes #332

## 💡 Additional Context

The `TripInviteItem` component now fetches the trip and user details asynchronously using `LaunchedEffect`. Placeholder values are shown until the data is retrieved. This improves user experience by eliminating the display of raw IDs.

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt [hunk 1]
```diff
@@ -550,4 +550,14 @@ open class TripsViewModel(
     tripInviteRepository.createTripInvite(
         req = tripInvite, onSuccess = onSuccess, onFailure = onFailure)
   }
+
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt:553]: Could you please add documentation for this function? Saves us some time for the M3 deadline.

File: app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt [hunk 2]
```diff
@@ -722,4 +722,14 @@ constructor(
             },
             onFailure = { exception -> Log.e("USER_VIEW_MODEL", exception.message.orEmpty()) })
   }
+
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt:725]: Same comment as above for this one :)

File: app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -166,17 +167,48 @@ fun ProfileContent(
...
+fun TripInviteItem(
+    tripInvite: TripInvite,
+    tripsViewModel: TripsViewModel,
+    userViewModel: UserViewModel
+) {
+  val tripName = remember { mutableStateOf("Loading trip...") }
+  val senderName = remember { mutableStateOf("Loading user...") }
+
+  // Fetch the trip and user details asynchronously
+  LaunchedEffect(tripInvite) {
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt:187]: Nice implementation to get the trip and user details!

File: app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt (hunk 1 above)
Comment (BogdanCDev) [reply app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt:553]: Thanks for pointing this out! Done!

File: app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt (hunk 2 above)
Comment (BogdanCDev) [reply app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt:725]: Done!

File: app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt (hunk 3 above)
Comment (BogdanCDev) [reply app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt:198]: Thank you very much!

**Review by 1andreea26 — APPROVED** [review]
Looks good Bogdan! Thank you for the quick fix and for adding documentation! Line coverage is 100%, very nice! I will approve.

**Pull request merged**